app/models/ip_based_group.rb

```diff
@@ -0,0 +1,41 @@
+class IpBasedGroup < ActiveRecord::Base
+  belongs_to :role
+  validate :has_valid_ip_addresses
+
+  def self.for_ip(ip)
+    ip_int = IpConverter.ip_to_i(ip)
+    if ip_int
```

Comment:
Normally I'd do something like this:

```
ip_int = ...
if !ip_int
  return []
end

return joins(:role)...
```

But I believe our style guide goes against multiple returns as well as explicit returns.

I could consider extracting to a specialized service like we did with `MaybeURI` or something, where have the if/else all in one place, and very simple functions for choosing the branching logic.